module/userProjects/helper.js

```diff
@@ -1048,323 +1046,353 @@ module.exports = class UserProjectsHelper {
-            //     if( projectDetails.length > 0 ) {
-            //         projectId = projectDetails[0]._id;
-            //     } else {
-                
-            //         let solutionDetails = {}
+            if (projectId === "") {
+                // This will check wether the user user is targeted to solution or not based on his userRoleInformation
+                const targetedSolutionId = await coreService.checkIfSolutionIsTargetedForUserProfile(userToken,userRoleInformation,solutionId)
+                //based on above api will check for projects wether its is private project or public project
+                const projectDetails = await projectQueries.projectDocument({
+                    solutionId: solutionId,
+                    userId: userId,
+                    isAPrivateProgram: targetedSolutionId.data.isATargetedSolution ? false : true
+                }, ["_id"]);
+                if( projectDetails.length > 0 ) {
+                    projectId = projectDetails[0]._id;
+                } else {
+                    let isAPrivateSolution = (targetedSolutionId.data.isATargetedSolution === false)  ? true : false;
+                    let solutionDetails = {}
+
+                    if( templateId === "" ) {
+                        // If solution Id of a private program is passed, fetch solution details
+                        if ( isAPrivateSolution && solutionId != "" ) {
+                            solutionDetails = await solutionsHelper.solutionDocuments({
+                                _id: solutionId,
+                                isAPrivateProgram: true
+                            },
+                            [
+                                "name",
+                                "externalId",
+                                "description",
+                                "programId",
+                                "programName",
+                                "programDescription",
+                                "programExternalId",
+                                "isAPrivateProgram",
+                                "projectTemplateId",
+                                "entityType",
+                                "entityTypeId",
+                                "language",
+                                "creator"
+                            ]);
+                            if( !solutionDetails.length > 0 ) {
+                                throw {
+                                    status : HTTP_STATUS_CODE["bad_request"].status,
+                                    message : CONSTANTS.apiResponses.SOLUTION_NOT_FOUND
+                                }
+                            }
+                            solutionDetails = solutionDetails[0];
+                        } else {
+                            solutionDetails = 
+                            await coreService.solutionDetailsBasedOnRoleAndLocation(
+                                userToken,
```

Comment:
@Prajwal17Tunerlabs 
why we are calling coreService ? Copy that API to project service